nifi-commons/nifi-flow-encryptor/src/test/java/org/apache/nifi/flow/encryptor/StandardFlowEncryptorTest.java

```diff
@@ -89,7 +100,145 @@ public void testProcessNoEncrypted() {
+        final Source test = Input.fromString(outputXml).build();
+        DifferenceEngine diff = new DOMDifferenceEngine();
+        diff.addDifferenceListener((comparison, outcome) -> {
+            final Matcher inputMatcher = PATTERN.matcher(comparison.getControlDetails().getValue().toString());
+            final Matcher outputMatcher = PATTERN.matcher(comparison.getTestDetails().getValue().toString());
+            if (inputMatcher.matches() && outputMatcher.matches()) {
+                assertEquals(inputEncryptor.decrypt(inputMatcher.group(1)), outputEncryptor.decrypt(outputMatcher.group(1)));
+            }
+        });
+        diff.compare(control, test);
+    }
+
     private PropertyEncryptor getPropertyEncryptor(final String propertiesKey, final String propertiesAlgorithm) {
         return new PropertyEncryptorBuilder(propertiesKey).setAlgorithm(propertiesAlgorithm).build();
     }
+
+    private String getSampleFlowJson(final String password) {
+        Objects.requireNonNull(password);
+        return String.format("{\n" +
+                "    \"processors\": [\n" +
+                "      {\n" +
+                "        \"name\": \"SAMPLE_PROCESSOR\",\n" +
+                "        \"comments\": \"\",\n" +
+                "        \"type\": \"org.apache.nifi.processors.SAMPLE_PROCESSOR\",\n" +
+                "        \"bundle\": {\n" +
+                "          \"group\": \"org.apache.nifi\",\n" +
+                "          \"artifact\": \"nifi-sample-bundle\"\n" +
+                "        },\n" +
+                "        \"properties\": {\n" +
+                "          \"Commit Offsets\": \"true\",\n" +
+                "          \"bootstrap.servers\": \"localhost:9092\",\n" +
+                "          \"topic_type\": \"names\",\n" +
+                "          \"max-uncommit-offset-wait\": \"1 secs\",\n" +
+                "          \"sasl.mechanism\": \"GSSAPI\",\n" +
+                "          \"honor-transactions\": \"true\",\n" +
+                "          \"message-header-encoding\": \"UTF-8\",\n" +
+                "          \"max.poll.records\": \"10000\",\n" +
+                "          \"security.protocol\": \"PLAINTEXT\",\n" +
+                "          \"sasl.token.auth\": \"false\",\n" +
+                "          \"separate-by-key\": \"false\",\n" +
+                "          \"username\": \"sample_username\",\n" +
+                "          \"password\": \"%s\",\n" +
+                "          \"sample_integer\": 10,\n" +
+                "          \"sample_float\": 2.5,\n" +
+                "          \"sample_true\": true,\n" +
+                "          \"sample_false\": false,\n" +
+                "          \"sample_empty_array\": [],\n" +
+                "          \"key-attribute-encoding\": \"utf-8\",\n" +
+                "          \"auto.offset.reset\": \"latest\"\n" +
+                "        }\n" +
+                "      }\n" +
+                "    ]\n" +
```

Comment:
This example contains a large number of unnecessary elements. Recommend reducing the example to just two properties.